cli/src/git_util.rs

```diff
@@ -172,17 +173,122 @@ pub fn with_remote_git_callbacks<T>(
     f(callbacks)
 }
 
-pub fn print_git_import_stats(ui: &mut Ui, stats: &GitImportStats) -> Result<(), CommandError> {
+pub fn print_git_import_stats(
+    ui: &mut Ui,
+    stats: &GitImportStats,
+    query_ref_stats: Option<&ReadonlyRepo>,
+) -> Result<(), CommandError> {
+    if let (Some(repo), true) = (query_ref_stats, !stats.changed_remote_refs.is_empty()) {
+        let mut remote_branches = Vec::new();
+        let mut tags = Vec::new();
+        let mut local_branches = Vec::new();
+
+        // group the references by type
+        for (refname, (remote_ref, ref_target)) in &stats.changed_remote_refs {
+            let ref_status = RefStatus::new(refname, remote_ref, ref_target, repo);
+            match refname {
+                RefName::RemoteBranch { .. } => remote_branches.push(ref_status),
+                RefName::Tag(_) => tags.push(ref_status),
+                RefName::LocalBranch(_) => local_branches.push(ref_status),
+            }
+        }
+
+        let grouped_refs = vec![remote_branches, tags, local_branches];
+
+        for refs in grouped_refs {
+            if let Some(max_ref_name_len) = refs.iter().map(|status| status.ref_name.len()).max() {
+                for status in refs {
+                    status.output(max_ref_name_len, &mut ui.stderr())?;
+                }
+                writeln!(ui.stderr())?;
+            }
+        }
+    }
+
     if !stats.abandoned_commits.is_empty() {
         writeln!(
             ui.stderr(),
             "Abandoned {} commits that are no longer reachable.",
             stats.abandoned_commits.len()
         )?;
     }
+
     Ok(())
 }
 
+struct RefStatus {
+    ref_name: String,
+    tracking_status: TrackingStatus,
+    import_status: ImportStatus,
+}
+
+impl RefStatus {
+    fn new(
+        ref_name: &RefName,
+        remote_ref: &RemoteRef,
+        ref_target: &RefTarget,
+        repo: &ReadonlyRepo,
+    ) -> Self {
+        let mut tracking_status = TrackingStatus::Untracked;
+        let ref_name = match ref_name {
+            RefName::RemoteBranch { branch, remote } => {
+                if repo.view().get_remote_branch(branch, remote).is_tracking() {
+                    tracking_status = TrackingStatus::Tracked;
+                }
+                format!("branch: {branch}@{remote}")
+            }
+            RefName::Tag(tag) => format!("tag: {tag}"),
+            RefName::LocalBranch(branch) => format!("branch: {branch}"),
+        };
+
+        let import_status = match (remote_ref.target.is_absent(), ref_target.is_present()) {
+            (true, true) => ImportStatus::New,
+            (false, false) => ImportStatus::Deleted,
+            _ => ImportStatus::Updated,
```

Comment:
nit: I would make both use `is_absent()` (or `is_present()`) so the table shows the difference `(false, true)` and `(true, false)`.

Reply:
resolved